### ISSUE TITLE
Request for multiplatform

### ISSUE THREAD

**Issue by NikkyAI** [nwillc/ksvg#3]
Is it possible to package this for multiplatform?

for my own project i just copied your code into my `commonMain`

i use this with kotlin/js and react, but having it in common code leaves the option open to render stuff serverside too

seems like Chat.isUpperCase() does not exist on js so i solved it expect/actual, iirc this is the only problem i ran into

```
//common
expect val Char.isUpperCase: Boolean

// js
actual val Char.isUpperCase: Boolean
    get() = (this == this.toUpperCase() && this != this.toLowerCase())

// jvm
actual val Char.isUpperCase: Boolean
    get() = isUpperCase()
```

**Comment by nwillc** [comment]
I've thought this would be a good thing to do.  I've not done any multi
platform yet. Will look at it.

**Comment by nwillc** [comment]
I assume the logging support will need fixing too.

**Comment by NikkyAI** [comment]
i would suggest kotlin-logging, not that there is much to replace..
have a look at my port: https://github.com/NikkyAI/pentagame/tree/a0311c68ade0dc768a539c6e8da6a8901682aa92/shared/src/commonMain/kotlin/com/github/nwillc/ksvg
and the Character.isUpperCase code here https://github.com/NikkyAI/pentagame/blob/ca4215cdade3e9f8b3be5519365620952dd4c9bd/shared/src/jsMain/kotlin/isUpperCase.kt

**Comment by nwillc** [comment]
I've undertaken making the project multi-platform.  Won't happen instantly,
and happy for any help, but I am working on it in my free time.

**Comment by nwillc** [comment]
I may have a working version. Would you be able to look at the
feature/multi-platform branch to review the JS work?  I don't have
expertise to evaluate the JS bits.

**Comment by NikkyAI** [comment]
i assume with js specific code you mean the logging?

it looks like it will work
but i think it will not show the original file location / line numbers when using the browser dev tools
for that iirc the logger calls need to all be inline or so, never looked into it in detail
korlibs/klogger works for me, but it has no decent formatting

also 1.3.70-eap had some mindblowing improvements to the js side of things
although it is fine for this, since you are depending on barely anything

through experimentation i found out that this or something similar is required in the buildscript

```
js() {
        useCommonJs()
        browser {
            runTask {
                sourceMaps = true
            }
            webpackTask {
                sourceMaps = true
            }
        }
        compilations.all {
            kotlinOptions {
                sourceMap = true
                metaInfo = true
//                outputFile = "${project.buildDir.path}/js/muirwik-components.js"
                main = "call"
//                moduleKind = "commonjs"
            }
        }
        mavenPublication { // Setup the publication for the target
//            artifactId = "ksvg-js"
            // Add a docs JAR artifact (it should be a custom task):
//            artifact(javadocJar)
        }
    }
```

without it the project using it will complain that ksvg is not configured for js
this error only nees to pop up when it is included as subproject, but after fixing it using it as a library from mavenLocal seems to work too

**Comment by nwillc** [comment]
I've accepted your PR.  It's not the logging I'm worried about. I don't have a JS test harness to try things with and am struggling getting JS unit testing going.

**Comment by NikkyAI** [comment]
i can look into that

also have similar problems of not knowing how to run tests in the browser

**Comment by NikkyAI** [comment]
seems like this is breaking things.. https://stackoverflow.com/questions/51475050/spaces-in-test-method-names-when-targeting-multiplatform

TL;DR: cannot have spaces in testnames in commonTest because it cannot compile to js

**Comment by NikkyAI** [comment]
after fixing all the testnames with `@JsName`

these errors are left over

here jvm and js seem to sort `x1`, `x2`, `y1`, `y2` differently
https://github.com/nwillc/ksvg/blob/72402ec00c1f4c33ec4f2d5f01335ef12000d207/src/commonTest/kotlin/com/github/nwillc/ksvg/elements/LINETest.kt#L48

`IllegalArgumentException: Value (10) is not a valid length or percentage`
https://github.com/nwillc/ksvg/blob/72402ec00c1f4c33ec4f2d5f01335ef12000d207/src/commonTest/kotlin/com/github/nwillc/ksvg/elements/RECTTest.kt#L43

`IllegalArgumentException: Value (M 10,30 A 20,20 0,0,1 50,30 Z) is not a valid path`
https://github.com/nwillc/ksvg/blob/72402ec00c1f4c33ec4f2d5f01335ef12000d207/src/commonTest/kotlin/com/github/nwillc/ksvg/elements/PATHTest.kt#L31

for the last few errors it seems like the regex might be acting up, i will open a PR again for the current progress

**Comment by nwillc** [comment]
The issues in the tests I believe are resolved.  The sort order is now enforced and the other bits and pieces seem resolved.  The JVM build artifacts now work with existing clients so I've merged the branch to master.  I think the javascript is working but I still don't understand the packaging and how it would be used in an example. More to do there.

**Comment by harry1453** [comment]
Please see #10 for instructions on how to use this in a multiplatform project
